maven-plugin-api/src/main/java/org/apache/maven/plugin/logging/Log.java

```diff
@@ -63,6 +65,20 @@
      */
     void debug( Throwable error );
 
+    /**
+     * Send a message to the user in the <b>debug</b> error level by computing the message
+     * only when needed. The supplier will be called only if @see #isDebugEnabled() is <b>true</b>.
+     * 
+     * @param messageSupplier a non null Supplier of the message to use
+     */
+    default void debug( Supplier<String> messageSupplier )
```

Comment:
don't think we need any default, I would add an AbstractLog our two impl would inherit from and that's it, will avoid the common default pitfalls - or corrollar is all our impl should impl it without using the default impl, up to you.

Reply:
I used the default methods not to break any other potential implementation.
If it is preferred not to do so then I can go with a breaking change and introduction of an Abstract class between the implementations. 
Let's wait a bit for others POV.

Reply:
We had that discussion on Artifact#setPath/getpath thread and it ended up as a compromise to do both.
I'm not sure it is worth but I'm fine with this but implementing it in implementation is the prerequisite on my side - not having it in the interface is fine for me since it is not a breaking change until you upgrade your mojo API and have a custom impl - which is fine.

Reply:
ok I'll remove the default methods from the interface

Reply:
default methods removed, abstract class introduced